[docs][joy] Add adjustments to the color inversion page (#37143)
mui/material-ui: docs/data/joy/main-features/color-inversion/color-inversion.md

{{"demo": "ColorInversionMotivation.js"}}

> On the **left**, the `Button`'s variant is `solid` which is the highest emphasis level compared to other components.

Reviewer: ```suggestion
On the **left**, the Button's variant is `solid`, which is the highest emphasis level compared to other components.
```

This conforms to the visual appearance on the screen.

On the **right**, the problem arises when the container's variant becomes `solid`.

> The button is no longer the highest emphasis element because the it has the same background as the container.

Reviewer: ```suggestion
The Button is no longer the highest emphasis element because the it has the same background as the container.
```

{{"demo": "ColorInversionOverview.js"}}

**Implicit** color refers to components that don't have the `color` prop specified.

> Color inversion has no effect on those children that have an **explicit** `color` prop.

Reviewer: Nitpick copy edit

```suggestion
Color inversion has no effect on children that have an **explicit** `color` prop.
```

## Usage

> To enable color inverstion, use the `invertedColors` prop on one of the two components that support it, either on the [`Sheet`](/joy-ui/react-sheet/) or the [`Card`](/joy-ui/react-card/).

Reviewer: ```suggestion
To enable color inversion, use the `invertedColors` prop on one of the two components that support it, either on the [Sheet](/joy-ui/react-sheet/) or the [Card](/joy-ui/react-card/).
```

> Make sure they are using either the `solid` or `soft` variants, as it will only work then.

Reviewer: Here's how I would phrase this:

```suggestion
Note that this prop only works when these components have the `solid` or `soft` global variants applied.
```

> To enable the feature set `invertedColors` to true on the surface components:

Reviewer: This line is redundant and can be cut.

```suggestion
```

```js
<Card variant="solid" invertedColors>…</Card>

<Sheet variant="soft" invertedColors>…</Sheet>
```

### Portal popup

> By default, color inversion has no effect on the popup slot of the `Autocomplete`, `Menu`, and `Tooltip` components.

Reviewer: ```suggestion
By default, color inversion has no effect on the popup slot of the Autocomplete, Menu, and Tooltip components.
```

To enable it, set `disablePortal` to true on the `slotProps`.

:::info

> The popup slot of the `Select` component has `disablePortal` set to true by default.

Reviewer: ```suggestion
The popup slot of the Select component has `disablePortal` set to true by default.
```

:::

{{"demo": "ColorInversionPopup.js"}}

## How it works

### Parent component

> When `invertedColors` is set to true on the surface component, a set of CSS variables are applied to it. The values of those variables comes from `theme.colorInversion[variant][color]` where `variant` and `color` are the component's props. The surface component also creates a React context to tell the children to update their styles.

Reviewer: ```suggestion
When `invertedColors` is set to true on the surface component, a set of CSS variables is applied to it. 
The values of those variables comes from `theme.colorInversion[variant][color]`, where `variant` and `color` are the component's props. 
The surface component also creates a React context to tell the children to update their styles.
```

## Usage

> To enable color inversion, use the `invertedColors` prop on one of the two components that support it, either on the [Sheet](/joy-ui/react-sheet/) or the [Card](/joy-ui/react-card/).

Reviewer: ```suggestion
To enable color inversion, use the `invertedColors` prop on one of the two components that support it: [Sheet](/joy-ui/react-sheet/) or [Card](/joy-ui/react-card/).
```

> Note that this prop only works when these components have the `solid` or `soft` global variants applied.

Reviewer: ```suggestion
Note that this prop only works when these components have the `solid` or `soft` global variant applied.
```

```js
<Card variant="solid" invertedColors>…</Card>

<Sheet variant="soft" invertedColors>…</Sheet>
```

### Portal popup

By default, color inversion has no effect on the popup slot of the Autocomplete, Menu, and Tooltip components.
To enable it, set `disablePortal` to true on the `slotProps`.

:::info
The popup slot of the Select component has `disablePortal` set to true by default.
:::

{{"demo": "ColorInversionPopup.js"}}

## How it works

### Parent component

When `invertedColors` is set to true on the surface component, a set of CSS variables is applied to it.

> The values of those variables comes from `theme.colorInversion[variant][color]`, where `variant` and `color` are the component's props.

Reviewer: ```suggestion
The values of those variables come from `theme.colorInversion[variant][color]`, where `variant` and `color` are the component's props.
```
